### ISSUE TITLE
syncd performWarmRestartSingleSwitch: failed to create switch RID: SAI_STATUS_INVALID_PARAMETER

### ISSUE THREAD

**Issue by vaibhavhd** [Azure/sonic-buildimage#6724]
**Description**
syncd warmstart (after issuing device warm-reboot) fails with `performWarmRestartSingleSwitch: failed to create switch RID: SAI_STATUS_INVALID_PARAMETER for VID oid:0x21000000000000`
This further leads to Orchagent crash: `meta_sai_on_switch_shutdown_request: switch_id oid:0x0 is of type SAI_OBJECT_TYPE_NULL, but expected SAI_OBJECT_TYPE_SWITCH`

Orchagent core is also generated as a result.

**Steps to reproduce the issue:**
1. Warm-reboot the DUT
2. Check syslog and core
3.

**Describe the results you received:**
```
Feb  8 17:07:54.803286 str2-7050cx3-acs-01 NOTICE admin: Rebooting with /sbin/kexec -e to SONiC-OS-HEAD.364-092f5378 ...
..
Feb  8 17:08:48.868899 str2-7050cx3-acs-01 INFO syncd#/supervisord: syncd init complete#015
..
..
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 ERR syncd#syncd: [none] SAI_API_SWITCH:_brcm_sai_db_table_node_list_init_v2:22818 table node list init failed with error -5.
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 ERR syncd#syncd: [none] SAI_API_ROUTER_INTERFACE:_brcm_sai_alloc_rif:3765 creating sub port vid list failed with error -5.
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 CRIT syncd#syncd: [none] SAI_API_SWITCH:_brcm_sai_dm_init:23277 initializing rif state failed with error -5.
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 CRIT syncd#syncd: [none] SAI_API_SWITCH:brcm_sai_create_switch:1563 initializing modules data failed with error -5.
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 NOTICE syncd#syncd: :- performWarmRestartSingleSwitch: Warm boot: create switch VID: oid:0x21000000000000 took 10.604614 sec
Feb  8 17:08:49.233755 str2-7050cx3-acs-01 ERR syncd#syncd: :- performWarmRestartSingleSwitch: failed to create switch RID: SAI_STATUS_INVALID_PARAMETER for VID oid:0x21000000000000
Feb  8 17:08:49.233903 str2-7050cx3-acs-01 NOTICE syncd#syncd: :- onSyncdStart: on syncd start took 10.615452 sec
Feb  8 17:08:49.233903 str2-7050cx3-acs-01 ERR syncd#syncd: :- run: Runtime error during syncd init: :- performWarmRestartSingleSwitch: failed to create switch RID: SAI_STATUS_INVALID_PARAMETER for VID oid:0x21000000000000
Feb  8 17:08:49.233953 str2-7050cx3-acs-01 NOTICE syncd#syncd: :- sendShutdownRequest: sending switch_shutdown_request notification to OA for switch: oid:0x0
Feb  8 17:08:49.234268 str2-7050cx3-acs-01 NOTICE syncd#syncd: :- sendShutdownRequestAfterException: notification send successfully
Feb  8 17:08:49.234352 str2-7050cx3-acs-01 NOTICE syncd#syncd: :- run: starting main loop, ONLY restart query
..
Feb  8 17:09:33.715450 str2-7050cx3-acs-01 ERR swss#orchagent: :- wait: SELECT operation result: TIMEOUT on notify
Feb  8 17:09:33.715613 str2-7050cx3-acs-01 ERR swss#orchagent: :- wait: failed to get response for notify
Feb  8 17:09:33.715732 str2-7050cx3-acs-01 ERR swss#orchagent: :- initSaiRedis: Failed to notify syncd INIT_VIEW, rv:-1 gSwitchId 0
Feb  8 17:09:33.715867 str2-7050cx3-acs-01 WARNING swss#orchagent: :- meta_sai_on_switch_shutdown_request: switch_id oid:0x0 is of type SAI_OBJECT_TYPE_NULL, but expected SAI_OBJECT_TYPE_SWITCH
Feb  8 17:09:33.715972 str2-7050cx3-acs-01 ERR swss#orchagent: :- operator(): not handled: SAI_OBJECT_TYPE_NULL
Feb  8 17:09:33.716709 str2-7050cx3-acs-01 NOTICE swss#orchagent: :- uninitialize: begin
Feb  8 17:09:33.716709 str2-7050cx3-acs-01 NOTICE swss#orchagent: :- uninitialize: begin
Feb  8 17:09:33.716757 str2-7050cx3-acs-01 NOTICE swss#orchagent: :- ~RedisChannel: join ntf thread begin
Feb  8 17:09:33.721496 str2-7050cx3-acs-01 INFO swss#/supervisord: orchagent terminate called after throwing an instance of 'std::runtime_error'
Feb  8 17:09:33.721496 str2-7050cx3-acs-01 INFO swss#/supervisord: orchagent   what():  :- operator(): not handled: SAI_OBJECT_TYPE_NULL
Feb  8 17:09:33.905687 str2-7050cx3-acs-01 INFO swss#supervisord 2021-02-08 17:09:33,904 INFO exited: orchagent (terminated by SIGABRT (core dumped); not expected)
Feb  8 17:09:34.919755 str2-7050cx3-acs-01 INFO swss#/supervisor-proc-exit-listener: Process 'orchagent' exited unexpectedly. Terminating supervisor 'swss'

Feb  8 17:09:39.342829 str2-7050cx3-acs-01 NOTICE root: Stopping swss service...
Feb  8 17:10:09.976688 str2-7050cx3-acs-01 NOTICE root: Starting swss service...
```

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

    **Output of `show version`:**

```
# show ver

SONiC Software Version: SONiC.HEAD.364-092f5378
Distribution: Debian 10.8
Kernel: 4.19.0-12-2-amd64
Build commit: 092f5378
Build date: Mon Feb  8 02:28:00 UTC 2021
Built by: johnar@jenkins-worker-22

Platform: x86_64-arista_7050cx3_32s
HwSKU: Arista-7050CX3-32S-C32
ASIC: broadcom
ASIC Count: 1
```

```
# docker exec -it syncd dpkg -s libsaibcm
Package: libsaibcm
Status: install ok installed
Priority: extra
Section: libs
Installed-Size: 232740
Maintainer: Guohan Lu <gulv@microsoft.com>
Architecture: amd64
Source: saibcm
Version: 4.3.0.10-4
```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

**Comment by vaibhavhd** [comment]
This issue is seen on TD3 platform and not seen on TH (S6100 T0) with same SONiC version on both:

```
Feb  8 16:35:09.917898 str-s6100-acs-2 NOTICE syncd#syncd: :- performWarmRestartSingleSwitch: switch oid:0x21000000000000
Feb  8 16:35:20.994678 str-s6100-acs-2 NOTICE syncd#syncd: :- performWarmRestartSingleSwitch: Warm boot: create switch VID: oid:0x21000000000000 took 11.070498 sec

Feb  8 16:35:21.684038 str-s6100-acs-2 NOTICE swss#orchagent: :- allocateNewSwitchObjectId: created SWITCH VID oid:0x21000000000000 for hwinfo: ''
Feb  8 16:35:21.688452 str-s6100-acs-2 NOTICE syncd#syncd: :- onSwitchCreateInInitViewMode: new switch oid:0x21000000000000 contains hardware info: ''
Feb  8 16:35:21.688626 str-s6100-acs-2 NOTICE syncd#syncd: :- onSwitchCreateInInitViewMode: current oid:0x21000000000000 switch hardware info: ''
Feb  8 16:35:21.688909 str-s6100-acs-2 NOTICE swss#orchagent: :- Switch: created switch with hwinfo = ''
```

```
SONiC Software Version: SONiC.HEAD.364-092f5378
Distribution: Debian 10.8
Kernel: 4.19.0-12-2-amd64
Build commit: 092f5378
Build date: Mon Feb  8 02:02:06 UTC 2021
Built by: johnar@jenkins-worker-22

Platform: x86_64-dell_s6100_c2538-r0
HwSKU: Force10-S6100
```

**Comment by vaibhavhd** [comment]
@gechiang this appears to be a new SAI issue in TD3 causing runtime error during syncd init and swss crash:

```
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 ERR syncd#syncd: [none] SAI_API_SWITCH:_brcm_sai_db_table_node_list_init_v2:22818 table node list init failed with error -5.
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 ERR syncd#syncd: [none] SAI_API_ROUTER_INTERFACE:_brcm_sai_alloc_rif:3765 creating sub port vid list failed with error -5.
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 CRIT syncd#syncd: [none] SAI_API_SWITCH:_brcm_sai_dm_init:23277 initializing rif state failed with error -5.
Feb  8 17:08:49.233679 str2-7050cx3-acs-01 CRIT syncd#syncd: [none] SAI_API_SWITCH:brcm_sai_create_switch:1563 initializing modules data failed with error -
```

**Comment by gechiang** [comment]
@vaibhavhd I tried the latest master image but it has problem with warmreboot precheck so that was not helpful.
Luckily I was able to go back to the same image that you used (364) and was able to reproduce the same issue.  It does looks like a SAI issue in SAI 4.3.0.10-4.  Since this master image can also run with the latest SAI 4.3.0.13-1, I tried loading it and repeated the warmreboot test and I see it was able to finish warm reboot without hitting this SAI issue any more.
Once master image overcomes the new warmreboot blocking issue we can try again to see if all is well and can proceed to close this issue if not seen again.
Thanks!
